tests/testthat/test-as_gt.R

```diff
@@ -0,0 +1,290 @@
+skip_on_cran()
+
+my_tbl_summary <- trial |>
+  select(trt, age, death) |>
+  tbl_summary() |>
+  modify_table_styling(columns = label, footnote = "test footnote", rows = variable == "age")
+my_tbl_cross <- tbl_cross(trial, trt, grade)
+my_tbl_regression <- lm(marker ~ age, trial) |> tbl_regression()
+my_tbl_uvregression <- trial |> tbl_uvregression(method = lm, y = age)
+my_spanning_tbl <- trial |>
+  tbl_summary(by = grade, include = age) |>
+  modify_spanning_header(c(stat_1, stat_3) ~ "**Testing**")
+
+expect_silent(gt_tbl_summary <- my_tbl_summary |> as_gt())
+expect_silent(gt_tbl_cross <- my_tbl_cross |> as_gt())
+expect_silent(gt_tbl_regression <- my_tbl_regression |> as_gt())
+expect_silent(gt_tbl_uvregression <- my_tbl_uvregression |> as_gt())
+expect_silent(gt_spanning_tbl <- my_spanning_tbl |> as_gt())
```

Comment:
If we want to reuse some of these and reduce duplication they can't be wrapped in a `test_that` chunk which stores all variables locally. I moved some of them which were not reused to inside their respective chunks, but what do you want to do about the others (e.g. `gt_tbl_summary` which is used in 4 tests)? I could just remove these lines and apply `as_gt` within each test as needed since I think `as_gt` is pretty quick... I have no strong opinion either way though I don't prefer having tests outside of a `test_that` chunk. Let me know what you think!

Reply:
Ahhh that makes sense. I don't have a strong opinion on it either. I guess my preference is to not wrap them in `expect_*()` fns at all. If they do fail, the testing procedure would fail too